include/PriorityQueue.h

```diff
@@ -4,16 +4,59 @@
 #include <list>
 #include <exception>
 
+/**
+ * @brief The MyComparator struct
+ * @details This struct is used to compare two objects of the same type.
+ * The operator() is used to compare two objects of the same type.
+ */
+template <class T>
+struct MyComparator {
+    T operator()(T a, T b) const {

```

Comment:
Notice that marking the arguments simply as T means they will be copied every call to the comparator. This is bad for performance and sometimes even impossible (i.e - if T is or contains a unique_ptr or any other type that deleted the Copy-constructor).
So in this case it’s definitely better to use "const-reference".
Check it out: https://www.geeksforgeeks.org/difference-between-call-by-value-and-call-by-reference/